orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pollers/RestorePinnedServerGroupsPoller.java

```diff
@@ -0,0 +1,316 @@
+import com.netflix.spectator.api.Counter;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.kork.core.RetrySupport;
+import com.netflix.spinnaker.orca.clouddriver.OortService;
+import com.netflix.spinnaker.orca.notifications.AbstractPollingNotificationAgent;
+import com.netflix.spinnaker.orca.pipeline.ExecutionLauncher;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionNotFoundException;
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository;
+import groovy.util.logging.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+import redis.clients.jedis.Jedis;
+import redis.clients.util.Pool;
+import retrofit.client.Response;
+import rx.Observable;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static com.netflix.spinnaker.orca.clouddriver.tasks.servergroup.PinnedServerGroupTagGenerator.PINNED_CAPACITY_TAG;
+
+@Slf4j
+@Component
+@ConditionalOnExpression(value = "${pollers.restorePinnedServerGroups.enabled:false}")
+class RestorePinnedServerGroupsPoller extends AbstractPollingNotificationAgent {
+  private static final Logger log = LoggerFactory.getLogger(RestorePinnedServerGroupsPoller.class);
+
+  private final ObjectMapper objectMapper;
+  private final OortService oortService;
+  private final RetrySupport retrySupport;
+  private final ExecutionLauncher executionLauncher;
+  private final ExecutionRepository executionRepository;
+
+  private final Counter errorsCounter;
+  private final Counter triggeredCounter;
+
+  @Autowired
+  public RestorePinnedServerGroupsPoller(@Qualifier("jedisPool") Pool<Jedis> jedisPool,
+                                         ObjectMapper objectMapper,
+                                         OortService oortService,
+                                         RetrySupport retrySupport,
+                                         Registry registry,
+                                         ExecutionLauncher executionLauncher,
+                                         ExecutionRepository executionRepository) {
+    super(jedisPool);
+
+    this.objectMapper = objectMapper;
+    this.oortService = oortService;
+    this.retrySupport = retrySupport;
+    this.executionLauncher = executionLauncher;
+    this.executionRepository = executionRepository;
+
+    this.errorsCounter = registry.counter("poller.restorePinnedServerGroups.errors");
```

Comment:
Not completely sold on these metric names ... thoughts?

Reply:
Would be nice to have all of these polling / agent / background jobs to have the same root name. Perhaps something like `job.{name}.{...}`? I'm fine with it otherwise.